ui/src/app/views/dataset-page.tsx

```diff
@@ -596,12 +607,15 @@ const DataSetPage = fp.flow(withCurrentWorkspace(), withUrlParams())(
                 based on the variable and value you selected above
               </div>
               </div>
-              <Button data-test-id='save-button'
-                onClick ={() => this.setState({openSaveModal: true})}
-                disabled={this.disableSave()}>
-                {this.editing ? !dataSetTouched ? 'Analyze' :
+              <TooltipTrigger data-test-id='save-tooltip'
+                           content='Requires Owner or Writer permission' disabled={this.canWrite}>
+                <Button data-test-id='save-button'
+                  onClick ={() => this.setState({openSaveModal: true})}
+                  disabled={this.disableSave() || !this.canWrite}>
```

Comment:
Opt: You might be able to fold the canWrite bit into disableSave, since I think the intent is the same?